src/snarks/groth16/mpc_utils.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef __ZETH_SNARKS_GROTH16_MPC_UTILS_HPP__
+#define __ZETH_SNARKS_GROTH16_MPC_UTILS_HPP__
+
+#include "snarks/groth16/powersoftau_utils.hpp"
+
+#include <vector>
+
+// Structures and utility functions related to CRS generation via an
+// MPC.
+
+namespace libzeth
+{
+
+using ppT = libff::default_ec_pp;
+
+/// Output from the second step of the first layer - the linear
+/// combination of elements in powersoftau, based on a specific
+/// circuit.
+class mpc_layer1
```

Comment:
What about renaming to `mpc_layer_result` maybe?

Reply:
Reamed to `srs_mpc_layer_L1` as discussed.